### PR TITLE
Refactor user manual, samples, and test configuration

### LINKED ISSUE #174: Documenting how to configure backups in the user manual
Once we have the core features of backup management in place, we should add an example of how to configure backups in the user manual.

### LINKED ISSUE #579: Building samples with Kustomize
Our sample cluster configurations involve a fair amount of repetition. This can lead to configuration drift where new features are not tested in all of the samples. I think we should explore building them with Kustomize, so we can define a base configuration and extend it with just the things that need to be customized for each sample case.

### LINKED ISSUE #710: Document FDB operator requirements fro TLS
We should document in our [user manual](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/docs/user_manual.md) the requirements for a TLS cluster setup. Currently users are not aware of the fact that they have to set the env variables and secret mount for the [operator](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/v0.35.0/controllers/pod_client.go#L86-L129):

```yaml
                env:
                  - name: FDB_TLS_CERTIFICATE_FILE
                    value: /tmp/fdb-certs/tls.crt
                  - name: FDB_TLS_CA_FILE
                    value: /tmp/fdb-certs/tls.crt
                  - name: FDB_TLS_KEY_FILE
                    value: /tmp/fdb-certs/tls.key
```

The error message is shown in: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/709 and we should probably think of printing out a more concrete error message e.g. `operator Pod is missing ...`.

### PR DESCRIPTION
**Pull request by brownleej**
These changes will hopefully make it easier for people to learn about using the operator, and give clearer guidance on configuration. This also expands the user manual to address major areas where we've seen questions recently.

Resolves #710 
Resolves #174 
Resolves #579

### CODE REVIEW

File: config/tests/backup/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+local development environment.
+
+The backup resource will bring up backup agents and start a backup. Once it
+is reconciled, you can run backup commands from the backup agent containers:
+
+    # Check Status
+    kubectl exec deployment/test-cluster-backup-agents -- fdbbackup status
+
+    # List Backups
+    kubectl exec deployment/test-cluster-backup-agents -- fdbbackup list -b "blobstore://minio@minio-service:9000?bucket=fdb-backups"
```
Comment (johscheuer) [inline config/tests/backup/README.md:11]:
```suggestion
```bash
# Check status
kubectl exec deployment/test-cluster-backup-agents -- fdbbackup status
```

```bash
# List backups
kubectl exec deployment/test-cluster-backup-agents -- fdbbackup list -b "blobstore://minio@minio-service:9000?bucket=fdb-backups"
```
```

File: config/tests/backup/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+9. Open a CLI and run `writemode on; clearrange '' \xff`.
+9. Confirm in the CLI that the test key is cleared.
+10. Apply the configuration for starting a restore: `kubectl apply -k config/tests/backup/restore`.
+11. Wait for all resources to be reconciled.
+12. Confirm through `fdbrestore status` that the restore has `State: completed`.
+13. Open a CLI and check the test key.
+
+Once that is done, you can clean up the backup by running:
+
+    kubectl exec deployment/test-cluster-backup-agents -- fdbbackup delete -d "blobstore://minio@minio-service:9000/test-cluster?bucket=fdb-backups"
```
Comment (johscheuer) [inline config/tests/backup/README.md:46]:
```suggestion
```bash
kubectl exec deployment/test-cluster-backup-agents -- fdbbackup delete -d "blobstore://minio@minio-service:9000/test-cluster?bucket=fdb-backups"
```
```

File: docs/manual/backup.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
+# Managing Backups through the Operator
+
+FoundationDB has out-of-the-box support for backing up to an S3-compatible object store, and the operator supports this through a special resource type for backup and restore. These backups run continuously, and simultaneously build new snapshots while backing up new mutations. You can restore to any point in time after the end of the first snapshot.
+
+You can find more information about the backup feature in the [FoundationDB Backup documentation](https://apple.github.io/foundationdb/backups.html).
+
+**Warning**: Support for backups in the operator is still in development, and there are significant missing features.
```
Comment (johscheuer) [inline docs/manual/backup.md:7]: Should we reference the GitHub project there?

File: docs/manual/fault_domains.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+spec:
+  version: 6.2.30
+  faultDomain:
+    key: topology.kubernetes.io/zone
+    valueFrom: spec.zoneName
+```
+
+The example above divides processes across nodes based on the label `topology.kubernetes.io/zone` on the node, and sets the zone locality information in FDB based on the field `spec.zoneName` on the pod. The latter field does not exist, so this configuration cannot work. There is no clear pattern in Kubernetes for allowing pods to access node information other than the host name, which presents challenges using any other kind of fault domain.
+
+If you have some other mechanism to make this information available in your pod's environment, you can tell the operator to use an environment varilable as the source for the zone locality:
```
Comment (johscheuer) [inline docs/manual/fault_domains.md:42]:
```suggestion
If you have some other mechanism to make this information available in your pod's environment, you can tell the operator to use an environment variable as the source for the zone locality:
```

**Review by johscheuer — COMMENTED** [review]
Overall that looks good and thanks for refactoring everything. I just wonder if we should/musst add the Apache license header to all files?

File: config/samples/backup.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# This sample assumes some configuration that you will need to fill in based on
+# the details of your environment. Those assumptions are explained in comments
+# below.
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBBackup
+metadata:
+  name: sample-cluster
+spec:
+  version: 6.2.30
+  clusterName: sample-cluster
```
[the remark this answers is not in the document]
Comment (brownleej) [reply config/samples/backup.yaml:12]: I'm not sure if we want to confine our usage of it in the docs. Is there an advantage to making that explicit?

[the remark this answers is not in the document]
Comment (johscheuer) [reply config/samples/backup.yaml:12]: I guess not. I just came across this since most other options are also commented.

**Review by johscheuer — COMMENTED** [review]
LGTM 👍

**Review by johscheuer — APPROVED** [review]

**Pull request merged**